### PR TITLE
Add kikkou keyboard

### PR DESCRIPTION
**Pull request by matthewdias**
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: keyboards/kikkou/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	    # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kikkou/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kikkou/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes		    # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no         # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/kikkou/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kikkou/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes		    # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no         # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/kikkou/rules.mk:21]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kikkou/kikkou.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT_all( \
```
Comment (noroadsleft) [inline keyboards/kikkou/kikkou.h:24]:
```suggestion
#define LAYOUT_65_ansi_blocker( \
```

File: keyboards/kikkou/info.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "Kikkou",
+    "url": "",
+    "maintainer": "Mechwerkes",
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (noroadsleft) [inline keyboards/kikkou/info.json:6]:
```suggestion
        "LAYOUT_65_ansi_blocker": {
```

File: keyboards/kikkou/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/default/keymap.c:22]:
```suggestion
	[0] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+		KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_HOME,  
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,  
+		KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_UP,   KC_PGDN,  
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_LALT, KC_LGUI, KC_LEFT, KC_DOWN, KC_RGHT
+	),
+
+	[1] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/default/keymap.c:30]:
```suggestion
	[1] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[2] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/default/keymap.c:38]:
```suggestion
	[2] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/default/keymap.c:46]:
```suggestion
	[3] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/via/keymap.c:22]:
```suggestion
	[0] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+		KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_HOME,  
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,  
+		KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_UP,   KC_PGDN,  
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_LALT, KC_LGUI, KC_LEFT, KC_DOWN, KC_RGHT
+	),
+
+	[1] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/via/keymap.c:30]:
```suggestion
	[1] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[2] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/via/keymap.c:38]:
```suggestion
	[2] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kikkou/keymaps/via/keymap.c:46]:
```suggestion
	[3] = LAYOUT_65_ansi_blocker(
```

File: keyboards/kikkou/kikkou.h [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+	K40, K41, K42,                K46,                K4A, K4B, K4C, K4D, K4E  \
+) { \
+	{ K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E }, \
+	{ K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E }, \
+	{ K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, XXX, K2D, K2E }, \
+	{ K30, XXX, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E }, \
+	{ K40, K41, K42, XXX, XXX, XXX, K46, XXX, XXX, XXX, K4A, K4B, K4C, K4D, K4E }  \
+}
+
+#define LAYOUT_tsangan( \
```
Comment (noroadsleft) [inline keyboards/kikkou/kikkou.h:38]:
```suggestion
#define LAYOUT_65_ansi_blocker_tsangan( \
```

File: keyboards/kikkou/info.json [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+        },
+        "LAYOUT_tsangan": {
```
Comment (noroadsleft) [inline keyboards/kikkou/info.json:9]:
```suggestion
        "LAYOUT_65_ansi_blocker_tsangan": {
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
* renames `LAYOUT_all` to `LAYOUT_65_ansi_blocker`
* renames `LAYOUT_tsangan` to `LAYOUT_65_ansi_blocker_tsangan`

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**